iModelCore/Units/src/Formatting/NumericFormatSpec.cpp

```diff
@@ -959,4 +963,46 @@ double NumericFormatSpec::RoundDouble(double dval, double roundTo)
     return (dval < 0.0) ? -rnd : rnd;
     }
 
+Utf8String NumericFormatSpec::FormatToFractionalRatio(double dval, Utf8String unitName) const
+{
+    double reciprocal = 1.0 / dval;
+
+    Utf8String result;
+    if (unitName == "VERTICAL_PER_HORIZONTAL")

```

Comment:
again, needs to work without hard coded units, the result should entirely depend on dval and nothing else. unitName should not be needed as a parameter